native/core/src/execution/datafusion/shuffle_writer.rs

```diff
@@ -1091,6 +1103,8 @@ async fn external_shuffle(
     partitioning: Partitioning,
     metrics: ShuffleRepartitionerMetrics,
     context: Arc<TaskContext>,
+    read_time: Time,
+    scan_time: Option<Time>,
```

Comment:
Why is `scan_time` an option?

Reply:
Just in case the child node of ShuffleWriterExec is not a ScanExec (I recall seeing cases where there is an intermediate CopyExec). In that case we could perhaps recurse down to find the ScanExec. CopyExec is another exec that we do not currently capture metrics for (https://github.com/apache/datafusion-comet/issues/935).

Reply:
this is no longer relevant